go-controller/pkg/node/openflowmanager/openflow_manager.go

```diff
@@ -0,0 +1,147 @@
+package openflowmanager
+
+import (
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/syncmap"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	"k8s.io/klog/v2"
+)
+
+// openflow manager per bridge
+type ofmanager struct {
+	bridge string
+	// flow cache, use map instead of array for readability when debugging
+	flowCache map[string][]string
+	flowMutex sync.Mutex
+	// channel to indicate we need to update flows immediately
+	flowChan chan struct{}
+}
+
+type Controller struct {
+	ofmanagers *syncmap.SyncMap[*ofmanager]
+	stopChan   <-chan struct{}
+}
+
+// NewController creates a new openflow manager
+func NewController(stopChan <-chan struct{}) *Controller {
+	return &Controller{
+		ofmanagers: syncmap.NewSyncMap[*ofmanager](),
+		stopChan:   stopChan,
+	}
+}
+
+// UpdateFlowCacheEntry update flow cache of the given bridge
+func (oc *Controller) UpdateFlowCacheEntry(bridgeName, key string, flows []string) {
+	_ = oc.ofmanagers.DoWithLock(bridgeName, func(bridge string) error {
+		c, found := oc.ofmanagers.LoadOrStore(bridge, &ofmanager{
+			bridge:    bridge,
+			flowCache: make(map[string][]string),
+			flowMutex: sync.Mutex{},
+			flowChan:  make(chan struct{}, 1),
+		})
+		c.flowMutex.Lock()
+		defer c.flowMutex.Unlock()
+		c.flowCache[key] = flows
+		// oc.stopChan is nil in unit testing, so it does not start syncFlow go-routine for unit tests.
```

Comment:
why is that ? Shouldn't we want to unit test something that resembles the live system ? 

Meaning, why can't we start the `syncFlow` go-routine for unit tests ?